src/jrd/obj.h

```diff
@@ -24,56 +24,89 @@
 #ifndef JRD_OBJ_H
 #define JRD_OBJ_H
 
-// Object types used in RDB$DEPENDENCIES and RDB$USER_PRIVILEGES
+// Object types used in RDB$DEPENDENCIES and RDB$USER_PRIVILEGES and stored in backup.
 // Note: some values are hard coded in grant.gdl
+// Keep existing constants unchanged.
 
-const int obj_relation			= 0;
-const int obj_view				= 1;
-const int obj_trigger			= 2;
-const int obj_computed			= 3;
-const int obj_validation		= 4;
-const int obj_procedure			= 5;
-const int obj_expression_index	= 6;
-const int obj_exception			= 7;
-const int obj_user				= 8;
-const int obj_field				= 9;
-const int obj_index				= 10;
-const int obj_charset			= 11;
-const int obj_user_group		= 12;
-const int obj_sql_role			= 13;
-const int obj_generator			= 14;
-const int obj_udf				= 15;
-const int obj_blob_filter		= 16;
-const int obj_collation			= 17;
-const int obj_package_header	= 18;
-const int obj_package_body		= 19;
-const int obj_privilege			= 20;
-
-const int obj_last_non_ddl		= 20;	// keep in sync!!!
-
-// objects types for ddl operations
-const int obj_database			= obj_last_non_ddl + 1;
-const int obj_relations			= obj_last_non_ddl + 2;
-const int obj_views				= obj_last_non_ddl + 3;
-const int obj_procedures		= obj_last_non_ddl + 4;
-const int obj_functions			= obj_last_non_ddl + 5;
-const int obj_packages			= obj_last_non_ddl + 6;
-const int obj_generators		= obj_last_non_ddl + 7;
-const int obj_domains			= obj_last_non_ddl + 8;
-const int obj_exceptions		= obj_last_non_ddl + 9;
-const int obj_roles				= obj_last_non_ddl + 10;
-const int obj_charsets			= obj_last_non_ddl + 11;
-const int obj_collations		= obj_last_non_ddl + 12;
-const int obj_filters			= obj_last_non_ddl + 13;
-
-const int obj_type_MAX			= obj_last_non_ddl + 14;	// keep this last!
-
-// used in the parser only / no relation with obj_type_MAX (should be greater)
-const int obj_user_or_role		= 100;
-const int obj_schema			= 101;
-const int obj_parameter			= 102;
-
-inline const char* get_object_name(int object_type)
```

Comment:
I doubt this change of object list to enum is needed. On the one hand it's correct change in good direction but on the other when that data is stored in database and used in for loops what we get is a lot of casts here & there, poisoning code. I like use of ObjectType but may be it should be not enum, just typedef? Or (if you want thin control on what's happening) create class ObjectType with appropriate auto casts, with checks in DEV_BUILD. Though must say I do not remember problems when illegal value of integer was used as object type...

Reply:
Done